### PR TITLE
refactor: restructure components by domain + prior color-token/test-rename cleanup

### PR DESCRIPTION
**Pull request by tunile943**
## Summary
- Restructures `app/components/` (and the parallel `styles/components/`) from atomic-design tiers (`atoms`/`molecules`/`organisms`) into domain folders: `calendar/`, `meeting-form/`, `admin/`, `navbar/`, `shared/`. `atoms/` is unchanged — it stays a flat, domain-agnostic tier since primitives get composed by every domain.
- `shared/` is deliberately small: only `FilterGroup` qualifies today (used by `calendar/MeetingsFilter`, `calendar/CompactCalendarSidebar`, and `admin/ExportTab`).
- Fixes a build-breaking bug surfaced while merging in previously-pushed work (see below): 7 `.module.scss` files referenced `$token` variables from `Variables.module.scss` without importing it — `tsc`/`eslint`/`stylelint` don't catch this since none of them run a real Sass compile, only `next build` does.
- Also includes two smaller refactors from a previous session that were pushed but never merged (`refactor/tests-rename-color-tokens`), now folded in here since they touch the same files: renaming `$dark-pink-color` → `$brand-pink-color`, replacing hardcoded hex values with their `Variables.module.scss` token equivalents, and renaming `frontend/test/` → `frontend/tests/` across configs/CI/docs.
- Updates `docs/03-development/project-structure.md` to describe the new domain layout.

## Test plan
- [x] `yarn tsc --noEmit`
- [x] `yarn lint` (0 errors, baseline warning count unchanged)
- [x] `yarn lint:css`
- [x] `yarn test:unit` (81 passed)
- [x] `yarn test:e2e` (72 passed)
- [x] `yarn build` (verifies every `.module.scss` import resolves — this is what caught the missing-import bug above)

## Summary by CodeRabbit

* **Style**
  * Refreshed UI theme styling to use consistent brand-pink and shared design variables across admin, calendar, meeting forms, navigation, and status controls.
  * Improved visual consistency for borders, separators, and indicators (including calendar overflow and Zoom host status states).
* **Documentation**
  * Updated project structure and testing/release checklist references to match the pluralized `tests` directory layout.
* **Tests**
  * Added/expanded end-to-end Playwright coverage for authentication, admin, meeting create/edit/delete, calendar views, filters, signage, Zoom/Google sync, recurring meetings, and edge cases.
  * Extended unit/integration coverage for route protection, sync behavior, recurrence, conflicts, and timezone utilities.

### CODE REVIEW

File: docs/03-development/testing/README.md [hunk 1]
```diff
@@ -7,16 +7,16 @@ onto the repo, this doc is the walkthrough; if you're about to sign off a releas
 
 ## The three tiers
 
-All test code lives under `frontend/test/`, split by tier:
+All test code lives under `frontend/tests/`, split by tier:
 
 ```
```
[the remark this answers is not in the document]
Comment (tunile943) [reply docs/03-development/testing/README.md]: Fixed in d941348 — added `text` as the fence language.

File: frontend/styles/components/calendar/WeeklyView.module.scss [hunk 2]
```diff
@@ -1,4 +1,5 @@
 @import 'https://fonts.googleapis.com/css2?family=Source+Sans+Pro:wght@400;600&display=swap';
+@import '../../Variables.module.scss';
```
[the remark this answers is not in the document]
Comment (tunile943) [reply frontend/styles/components/calendar/WeeklyView.module.scss:2]: Skipping this one: the repo's `config/stylelint.config.mjs` explicitly sets `scss/load-partial-extension: "always"` (with a comment) — keeping the `.scss` extension on partial imports is the intentional codebase convention here, the opposite of this rule's default. Removing it would go against that and isn't actually flagged by our stylelint run (`yarn lint:css` passes clean as-is).
